pdns/validate.cc

```diff
@@ -952,6 +982,64 @@ void validateDNSKeysAgainstDS(time_t now, const DNSName& zone, const dsmap_t& ds
       //        if(validkeys.empty()) cerr<<"did not manage to validate DNSKEY set based on DS-validated KSK, only passing KSK on"<<endl;
     }
   }
+
+  if (validkeys.size() < tkeys.size()) {
+    /* so we failed to validate the whole set, let's try to find out why exactly */
```

Comment:
What if one key is bad but the rest are good? do we still bail with a Bogus?

Reply:
You mean if we did not find at least one DNSKEY, corresponding to a DS, that validates the signature of the whole DNSKEY RRset? Yes